# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxFont.h
# Commit: 4c3cb20a5203
# Full Hash: 4c3cb20a52034b3cead6ac88e008f17bd446fb1c
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2022-08-16 16:01:29
# Regressor Bug: 1740530
# File Overlap Count: 1
# Description:
#   Bug 1740530 - patch 7 - Route COLRv1 fonts on macOS through our COLR-rendering path; only COLRv0 can use the native Core Text support. r=gfx-reviewers,lsalzman
#   
#   Depends on D153880
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152162
# ==============================================================================

diff -r 275eb6c4900b -r 4c3cb20a5203 gfx/thebes/gfxFont.h
--- a/gfx/thebes/gfxFont.h	Tue Aug 16 10:39:52 2022 +0000
+++ b/gfx/thebes/gfxFont.h	Tue Aug 16 10:39:52 2022 +0000
@@ -2255,6 +2255,11 @@
                         mozilla::gfx::DrawOptions drawOptions,
                         const mozilla::gfx::Point& aPoint, uint32_t aGlyphId);
 
+  // Subclasses can override to return true if the platform is able to render
+  // COLR-font glyphs directly, instead of us painting the layers explicitly.
+  // (Currently used only for COLR.v0 fonts on macOS.)
+  virtual bool UseNativeColrFontSupport() const { return false; }
+
   // Bug 674909. When synthetic bolding text by drawing twice, need to
   // render using a pixel offset in device pixels, otherwise text
   // doesn't appear bolded, it appears as if a bad text shadow exists